cmake/Config.cmake

```diff
@@ -122,6 +131,10 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
     if(GCC_MACHINE MATCHES ".*w64.*")
         set(SFML_COMPILER_GCC_W64 1)
     endif()
+
+    if(GCC_COMPILER_VERSION MATCHES "ucrt")
+        message(WARNING "The pre-compiled SFML dependencies for Windows are not compatible with the UCRT. Either switch to the MSVCRT or build the dependencies yourself.")
```

Comment:
Do we want to take this a step further and emit a fatal error when `SFML_USE_SYSTEM_DEPS` is false and then emit no warnings when it's true? That way those trying to link to our precompiled deps get an error they cannot miss while those using system deps are not annoyed with any warning?

Reply:
```cmake
if(GCC_COMPILER_VERSION MATCHES "ucrt" AND NOT SFML_USE_SYSTEM_DEPS)
    message(FATAL_ERROR "The pre-compiled SFML dependencies for Windows are not compatible with the UCRT. Either switch to the MSVCRT or build the dependencies yourself.")
endif()
```

This is what I had in mind.

I also think the error message ought to mention MinGW as to not be confused with the binaries that MSVC or Clang use.

Reply:
As we don't really have access to `SFML_USE_SYSTEM_DEPS` yet in the Config.cmake, I moved it one level higher after the `SFML_USE_SYSTEM_DEPS` check. That way it's also kind of grouped.